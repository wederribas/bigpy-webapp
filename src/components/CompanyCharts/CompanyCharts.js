import React from 'react'
import PropTypes from 'prop-types'
import { Grid, Paper, withStyles } from '@material-ui/core'
import ChartWrapper from '../ChartWrapper/ChartWrapper'

const styles = theme => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
    marginTop: theme.spacing.unit * 2
  }
})

function CustomerReportsPerYearChart(props) {
  const { data } = props

  const years = data.map(entry => entry.year)
  const countPerYear = data.map(entry => entry.count)

  const chartData = {
    labels: years,
    datasets: [
      {
        backgroundColor: 'rgba(213,108,104,0.2)',
        borderColor: 'rgba(213,108,104,1)',
        borderWidth: 1,
        pointHoverBackgroundColor: 'rgba(213,108,104,0.4)',
        pointHoverBorderColor: 'rgba(213,108,104,1)',
        data: countPerYear
      }
    ]
  }

  const options = {
    legend: {
      display: false
    },
    title: {
      display: true,
      text: 'Reports per Year'
    }
  }

  return <ChartWrapper data={chartData} options={options} type="line" />
}

CustomerReportsPerYearChart.propTypes = {
  data: PropTypes.array.isRequired
}

function CustomerReportsPerRegionChart(props) {
  const { data } = props

  const regions = data.map(entry => entry.region)
  const countPerRegion = data.map(entry => entry.count)

  const chartData = {
    datasets: [
      {
        data: countPerRegion,
        backgroundColor: [
          'rgba(93,172,163)',
          'rgba(100,190,121)',
          'rgba(238,203,116)',
          'rgba(203,64,57)',
          'rgba(70,131,195)'
        ],
        label: 'Dataset 1'
      }
    ],
    labels: regions
  }

  const options = {
    legend: {
      position: 'bottom'
    },
    title: {
      display: true,
      text: 'Reports per Region'
    }
  }

  return <ChartWrapper data={chartData} options={options} type="doughnut" />
}

CustomerReportsPerRegionChart.propTypes = {
  data: PropTypes.array.isRequired
}

function CustomerReportsRatingPerYearChart(props) {
  const { data } = props

  const years = data.map(entry => entry.year)
  const averagePerYear = data.map(entry => Number(entry.average).toFixed(2))

  const chartData = {
    labels: years,
    datasets: [
      {
        borderColor: 'rgba(70,131,195)',
        borderWidth: 2,
        pointHoverBackgroundColor: 'rgba(70,131,195,0.4)',
        pointHoverBorderColor: 'rgba(70,131,195,1)',
        data: averagePerYear,
        fill: false
      }
    ]
  }

  const options = {
    legend: {
      display: false
    },
    title: {
      display: true,
      text: 'Rating per Year'
    }
  }

  return <ChartWrapper data={chartData} options={options} type="line" />
}

CustomerReportsRatingPerYearChart.propTypes = {
  data: PropTypes.array.isRequired
}

function CustomerReportPerGenderChart(props) {
  const { data } = props

  const years = ['Female', 'Male']
  const genderCount = data.map(entry => entry.count)

  const chartData = {
    labels: years,
    datasets: [
      {
        backgroundColor: ['rgba(93,172,163,0.2)', 'rgba(100,190,121,0.2)'],
        borderColor: ['rgba(93,172,163,1)', 'rgba(100,190,121,1)'],
        borderWidth: 1,
        pointHoverBackgroundColor: 'rgba(213,108,104,0.4)',
        pointHoverBorderColor: 'rgba(213,108,104,1)',
        data: genderCount
      }
    ]
  }

  const options = {
    legend: {
      display: false
    },
    title: {
      display: true,
      text: 'Reports per Gender'
    }
  }

  return <ChartWrapper data={chartData} options={options} type="bar" />
}

CustomerReportPerGenderChart.propTypes = {
  data: PropTypes.array.isRequired
}

function CustomerReportPerAgeChart(props) {
  const { data } = props

  const years = data.map(entry => entry.ageRange)
  const ageRangeCount = data.map(entry => entry.count)

  const chartData = {
    labels: years,
    datasets: [
      {
        data: ageRangeCount,
        backgroundColor: [
          'rgba(93,172,163)',
          'rgba(100,190,121)',
          'rgba(238,203,116)',
          'rgba(203,64,57)',
          'rgba(70,131,195)',
          'rgba(23,40,56)',
          'rgba(50,99,94)'
        ],
        label: 'Dataset 1'
      }
    ]
  }

  const options = {
    legend: {
      display: true,
      position: 'bottom'
    },
    title: {
      display: true,
      text: 'Reports per Age Range'
    }
  }

  return <ChartWrapper data={chartData} options={options} type="polarArea" />
}

CustomerReportPerAgeChart.propTypes = {
  data: PropTypes.array.isRequired
}

function CompanyCharts(props) {
  const {
    classes,
    data: {
      reportsPerDate,
      companyReportsPerRegion,
      companyRatingPerYear,
      companyReportsPerGender,
      companyReportsPerAge
    }
  } = props

  return (
    <Paper elevation={1} className={classes.root}>
      <Grid container spacing={24}>
        <Grid item lg={6} md={12} xs={12}>
          <CustomerReportsPerYearChart data={reportsPerDate} />
        </Grid>
        <Grid item lg={6} md={6} xs={12}>
          <CustomerReportsRatingPerYearChart data={companyRatingPerYear} />
        </Grid>
        <Grid item lg={6} md={6} xs={12}>
          <CustomerReportPerGenderChart data={companyReportsPerGender} />
        </Grid>
        <Grid item lg={6} md={12} xs={12}>
          <CustomerReportsPerRegionChart data={companyReportsPerRegion} />
        </Grid>
        <Grid item lg={12} md={12} xs={12}>
          <CustomerReportPerAgeChart data={companyReportsPerAge} />
        </Grid>
      </Grid>
    </Paper>
  )
}

CompanyCharts.propTypes = {
  classes: PropTypes.object.isRequired,
  data: PropTypes.object.isRequired
}

export default withStyles(styles)(CompanyCharts)
