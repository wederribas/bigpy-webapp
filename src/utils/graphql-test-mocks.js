import { getCompaniesQuery } from '../components/Search/Search'

export const mocks = [
  {
    request: {
      query: getCompaniesQuery
    },
    result: {
      data: {
        allCompaniesNames: {
          edges: [
            { node: { Id: 'Company 0' } },
            { node: { Id: 'Company 1' } },
            { node: { Id: 'Company 2' } }
          ]
        }
      }
    }
  }
]
