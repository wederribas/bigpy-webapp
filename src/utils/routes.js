import Search from '../components/Search/Search'
import Top10 from '../components/Top10/Top10'
import Insights from '../components/Insights/Insights'
import About from '../components/About/About'
import Dashboard from '../components/Dashboard/Dashboard'

export const routes = [
  {
    path: '/',
    component: Search,
    exact: true
  },
  {
    path: '/dashboard/:companyName',
    component: Dashboard
  },
  {
    path: '/top10',
    component: Top10
  },
  {
    path: '/insights',
    component: Insights
  },
  {
    path: '/about',
    component: About
  }
]
