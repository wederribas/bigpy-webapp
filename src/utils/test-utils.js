import React from 'react'
import { render } from 'react-testing-library'
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'

const theme = createMuiTheme({
  typography: {
    useNextVariants: true
  }
})

const customRender = node => {
  return render(<MuiThemeProvider theme={theme}>{node}</MuiThemeProvider>)
}

export * from 'react-testing-library'

export { customRender as render }
