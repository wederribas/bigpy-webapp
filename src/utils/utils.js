// @flow

/**
 * Deburr special Latin letters to basic Latin letters and removing
 * combining diacritical marks.
 */
export function deburr(string: string): string {
  return string.normalize('NFD').replace(/[\u0300-\u036f]/g, '')
}
