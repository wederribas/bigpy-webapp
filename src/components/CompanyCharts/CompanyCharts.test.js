import React from 'react'
import { render } from '../../utils/test-utils'
import CompanyCharts from './CompanyCharts'

const chartsData = {
  reportsPerDate: [
    {
      year: '2018',
      count: '541'
    }
  ],
  companyReportsPerRegion: [
    {
      count: '61406',
      region: 'SE'
    },
    {
      count: '24333',
      region: 'NE'
    }
  ],
  companyRatingPerYear: [
    {
      average: '2.77',
      year: '2014'
    },
    {
      average: '2.51',
      year: '2015'
    }
  ],
  companyReportsPerGender: [
    {
      count: '34509',
      gender: 'F'
    },
    {
      count: '78990',
      gender: 'M'
    }
  ],
  companyReportsPerAge: [
    {
      ageRange: '0-20',
      count: '8289'
    },
    {
      ageRange: '61-70',
      count: '878'
    }
  ]
}

test('<CompanyCharts /> component renders component', () => {
  const { container } = render(<CompanyCharts data={chartsData} />)

  expect(container.firstChild).toMatchSnapshot()
})
