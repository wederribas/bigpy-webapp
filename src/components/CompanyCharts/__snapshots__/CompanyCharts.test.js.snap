// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompanyCharts /> component renders component 1`] = `
<div
  class="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 CompanyCharts-root-1"
>
  <div
    class="MuiGrid-container-29 MuiGrid-spacing-xs-24-53"
  >
    <div
      class="MuiGrid-item-30 MuiGrid-grid-xs-12-69 MuiGrid-grid-md-12-97 MuiGrid-grid-lg-6-105"
    >
      <canvas />
    </div>
    <div
      class="MuiGrid-item-30 MuiGrid-grid-xs-12-69 MuiGrid-grid-md-6-91 MuiGrid-grid-lg-6-105"
    >
      <canvas />
    </div>
    <div
      class="MuiGrid-item-30 MuiGrid-grid-xs-12-69 MuiGrid-grid-md-6-91 MuiGrid-grid-lg-6-105"
    >
      <canvas />
    </div>
    <div
      class="MuiGrid-item-30 MuiGrid-grid-xs-12-69 MuiGrid-grid-md-12-97 MuiGrid-grid-lg-6-105"
    >
      <canvas />
    </div>
    <div
      class="MuiGrid-item-30 MuiGrid-grid-xs-12-69 MuiGrid-grid-md-12-97 MuiGrid-grid-lg-12-111"
    >
      <canvas />
    </div>
  </div>
</div>
`;
